### PR TITLE
Add java.util.EventObject

### PR DESCRIPTION
**Pull request by Atry**
The class is part of Java Bean specification, and used by [Binding.scala](https://github.com/ThoughtWorksInc/Binding.scala)

### CODE REVIEW

**Comment by gzm0** [comment]
Jenkins, OK to test

File: javalib/src/main/scala/java/util/EventObject.scala [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package java.util
+
+class EventObject(@transient protected val source: AnyRef) extends java.io.Serializable {
```
Comment (sjrd) [inline javalib/src/main/scala/java/util/EventObject.scala]: Don't use `@transient`

File: javalib/src/main/scala/java/util/EventObject.scala [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+  private val serialVersionUID: Long = 5516075349620653480L
```
Comment (sjrd) [inline javalib/src/main/scala/java/util/EventObject.scala]: Do not define `serialVersionUID`. It's not a thing in Scala.js.

**Comment by sjrd** [comment]
There should be tests to assert that the methods and fields link properly.

**Comment by sjrd** [comment]
@nicolasstucki Could you review wrt the JavaDoc?

File: javalib/src/main/scala/java/util/EventObject.scala [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package java.util
+
+class EventObject(protected val source: AnyRef) extends java.io.Serializable {
+  def getSource = source
```
Comment (nicolasstucki) [inline javalib/src/main/scala/java/util/EventObject.scala]: There should be a `protected var source: AnyRef` field.

Comment (sjrd) [inline javalib/src/main/scala/java/util/EventObject.scala]: Also `getSource` should have `()` and an explicit result type.

Comment (nicolasstucki) [inline javalib/src/main/scala/java/util/EventObject.scala]: Add the type to all public members and parentesis to the methods. 
```scala
def getSource(): AnyRef = source
```

Comment (nicolasstucki) [inline javalib/src/main/scala/java/util/EventObject.scala]: You should also implement `toString`. The format look like 
```scala
this.getClass.getName + "[source=" + source.toString + ']'
```

**Comment by nicolasstucki** [comment]
That is all.

**Comment by nicolasstucki** [comment]
Still missing the `toString`

**Comment by nicolasstucki** [comment]
Jenkins, test this please.

**Comment by sjrd** [comment]
Repeating myself:

> There should be tests to assert that the methods and fields link properly.

**Comment by sjrd** [comment]
Ping @Atry ?

**Comment by Atry** [comment]
I have added the `toString` method. Unfortunately I still have no idea about how to test it yet.

**Comment by sjrd** [comment]
See for example https://github.com/scala-js/scala-js/commit/c764c1fc21652b7feae2801325159c9e218883bd or https://github.com/scala-js/scala-js/commit/b5ab6a34f46d32f0b80876ca22d00a77b1c22481 for how to add tests about the javalib.

**Comment by sjrd** [comment]
Jenkins, retest this please

File: javalib/src/main/scala/java/util/EventObject.scala [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package java.util
+
+class EventObject(protected var source: AnyRef) extends java.io.Serializable {
+
```
Comment (sjrd) [inline javalib/src/main/scala/java/util/EventObject.scala]: CI says: whitespace at EOL

File: javalib/src/main/scala/java/util/EventObject.scala [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package java.util
+
+class EventObject(protected var source: AnyRef) extends java.io.Serializable {
+  
+  def getSource(): AnyRef = source
+
```
Comment (sjrd) [inline javalib/src/main/scala/java/util/EventObject.scala]: CI says: whitespace at EOL

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/EventObjectTest.scala [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    }
+  }
+
+  @Test def shouldHaveGetSourceMethod(): Unit = {
+    import java.util._
+    val mySource = new AnyRef
+    assertEquals(mySource, new EventObject(mySource).getSource)
+  }
+
+
```
Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/EventObjectTest.scala]: Double blank line.

Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/EventObjectTest.scala]: Also add a test for `toString`

**Comment by sjrd** [comment]
When you're done, please squash all the commits into a single commit.

**Comment by sjrd** [comment]
Superseded by #2533.

**Pull request closed without merging**